drivers: regulator: Place API in iterable section

pdgendt opened the pull request:
Add wrapper DEVICE_API macro to all regulator_driver_api instances.

pdgendt committed:
drivers: regulator: Place API in iterable section

Add wrapper DEVICE_API macro to all regulator_driver_api instances.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

pdgendt committed:
drivers: regulator: parent: Place API into iterable section

Add wrapper DEVICE_API macro to all regulator_parent_driver_api instances.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

gmarull reviewed (changes requested):
some tests are deleted, but they need a solution (at least a workaround in test environment)

pdgendt committed:
tests: drivers: regulator: Test not implemented APIs with dummies

Instead of writing to existing fake APIs, use dummy devices with empty
APIs. This allows us to place APIs in ROM.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

pdgendt commented:
> some tests are deleted, but they need a solution (at least a workaround in test environment)

Added dummy devices and kept the tests.
